src/components/tools/pen.js

```diff
@@ -256,106 +276,114 @@ AFRAME.registerComponent("pen", {
-
-            this.el.object3D.matrixNeedsUpdate = true;
-            this.worldPosition.copy(this.intersection.point);
-          } else {
-            this.el.object3D.position.copy(this.originalPosition);
-            this.el.object3D.matrixNeedsUpdate = true;
-            getLastWorldPosition(this.el.object3D, this.worldPosition);
-          }
+          return rawIntersections[0];
         }
+      }
+
+      return null;
+    };
+  })(),
+
+  _updatePenTip(intersection) {
+    if (this.data.drawMode === DRAW_MODE.PROJECTION) {
+      if (intersection) {
+        this.el.object3D.position.copy(intersection.point);
+
+        this.el.parentEl.object3D.worldToLocal(this.el.object3D.position);
+
+        this.el.object3D.matrixNeedsUpdate = true;
+        this.worldPosition.copy(intersection.point);
       } else {
         this.el.object3D.position.copy(this.originalPosition);
         this.el.object3D.matrixNeedsUpdate = true;
         getLastWorldPosition(this.el.object3D, this.worldPosition);
       }
+    } else {
+      this.el.object3D.position.copy(this.originalPosition);
+      this.el.object3D.matrixNeedsUpdate = true;
+      getLastWorldPosition(this.el.object3D, this.worldPosition);
+    }
+  },
 
-      this._setPenVisible(this.grabberId !== "cursor" || !this.intersection);
-      this._setLineVisible(
-        this.el.sceneEl.is("vr-mode") && this.drawMode === DRAW_MODE.PROJECTION && this.intersection
-      );
-
-      //Prevent drawings from "jumping" large distances
-      if (
-        this.currentDrawing &&
-        (this.lastIntersectedObject !== (this.intersection ? this.intersection.object : null) &&
-          (!this.intersection ||
-            Math.abs(this.intersection.distance - this.lastIntersectionDistance) > MAX_DISTANCE_BETWEEN_SURFACES))
-      ) {
-        this.worldPosition.copy(this.lastPosition);
-        this._endDraw();
+  _updateLaser: (() => {
+    const laserStartPosition = new THREE.Vector3();
+    const laserEndPosition = new THREE.Vector3();
+    const camerWorldPosition = new THREE.Vector3();
+    const remoteLaserOrigin = new THREE.Vector3();
+    return function(cursorPose, intersection) {
+      if (cursorPose) {
+        laserStartPosition.copy(cursorPose.position);
+      } else {
+        this.el.parentEl.object3D.getWorldPosition(laserStartPosition);
       }
-      this.lastIntersectionDistance = this.intersection ? this.intersection.distance : 0;
-      this.lastIntersectedObject = this.intersection ? this.intersection.object : null;
 
-      if (!almostEquals(0.005, this.worldPosition, this.lastPosition)) {
-        this.direction.subVectors(this.worldPosition, this.lastPosition).normalize();
-        this.lastPosition.copy(this.worldPosition);
+      laserEndPosition.copy(intersection.point);
+
+      if (this.el.sceneEl.is("vr-mode")) {
+        remoteLaserOrigin.copy(laserStartPosition);
+      } else {
+        camerWorldPosition.copy(this.data.camera.object3D.position);
```

Comment:
Looks like `cameraWorldPosition` is immediately overwritten in the next line.

Reply:
Ah yea, i think that used to be `worldToLocal`, thanks